crates/subspace-farmer/src/plotting.rs

```diff
@@ -0,0 +1,379 @@
+use crate::commitments::Commitments;
+use crate::identity::Identity;
+use crate::object_mappings::ObjectMappings;
+use crate::plot::Plot;
+use crate::rpc::{EncodedBlockWithObjectMapping, FarmerMetadata, RpcClient};
+use anyhow::{anyhow, Result};
+use log::{debug, error, info};
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+use subspace_archiving::archiver::{ArchivedSegment, BlockArchiver, ObjectArchiver};
+use subspace_archiving::pre_genesis_data;
+use subspace_core_primitives::objects::{GlobalObject, PieceObject, PieceObjectMapping};
+use subspace_core_primitives::Sha256Hash;
+use subspace_solving::SubspaceCodec;
+use tokio::{sync::mpsc, task::JoinHandle};
+
+pub struct Plotting {
+    sender: Option<mpsc::Sender<()>>,
+    handle: Option<JoinHandle<Result<()>>>,
+}
+
+impl Plotting {
+    pub fn start(
+        plot: Plot,
+        commitments: Commitments,
+        object_mappings: ObjectMappings,
+        client: RpcClient,
+        identity: Identity,
+    ) -> Self {
+        let (sender, receiver) = mpsc::channel(1);
+
+        let plotting_handle = tokio::spawn(async move {
+            background_plotting(
+                client,
+                plot,
+                commitments,
+                object_mappings,
+                &identity.public_key(),
+                receiver,
+            )
+            .await
+        });
+
+        Plotting {
+            sender: Some(sender),
+            handle: Some(plotting_handle),
+        }
+    }
+
+    pub async fn wait(mut self) -> Result<()> {
```

Comment:
Being a library API, ideally, we wouldn't use `anyhow` here and in the other module

Reply:
Ok, will do